### ISSUE TITLE
Invalid code generation when using the question mark operator directly after `.await`

### ISSUE THREAD

**Issue by schitcrafter** [evcxr/evcxr#356]
## Description

When trying to execute this piece of code, evcxr generates invalid code in [lib.rs](https://github.com/evcxr/evcxr/files/13563197/lib.rs.txt) (`let _ = let mut mutex = ...`), leading to the error `expected expression, found "let" statement`.

```
:dep tokio = { version = "1.20.1", features = ["full"] }

let test = async { Ok::<_, &str>("") }.await?;
```

## Local setup

evcxr v0.16.0, rust 1.74.0, Windows 11

**Comment by PartiallyTyped** [comment]
Seems to work for me.

```
rustc 1.77.0-nightly (d78329b92 2024-01-13)
binary: rustc
commit-hash: d78329b92e8d141d19505e7c1527181c4ab87ed4
commit-date: 2024-01-13
host: aarch64-apple-darwin
release: 1.77.0-nightly
LLVM version: 17.0.6
```

**Comment by davidlattimore** [comment]
Thanks for reporting this @schitcrafter. I fixed this a couple of days after you reported it, but forgot to close this issue. The fix should be in 0.17.0.
